Fixed the GruntFile so it works now

drew-wallace opened the pull request:
I was gettings some issues with the SQLite package and the grunt build. I fixed it.

Commit:
Now it actually works

thomaswilburn commented:
I think rather than load two Grunt plugins, I'd rather just find out what was causing the hand-written `browserify` task (which is still in your Gruntfile) to crash. Especially if it's related to the SQLite, which Weir doesn't actually use. Can you provide more detail?

thomaswilburn commented on Hound.js:
```diff
@@ -194,24 +194,23 @@ var getMeta = function(url, c) {
-      });
+  });
+
+  var parser = new FeedParser();
```
This seems like a significant change. The previous code only creates a parser when the request is successful--this one creates it even if the request errored out. I'm not sure that instantiating the parser is expensive, but I don't see any reason not to be lazy with it, unless this was breaking something for your feeds.

drew-wallace commented:
I am on Node 5.7. When I tried to `npm install` I got an error referencing the SQLite3 package. Updating the SQLite version in package.json fixed it, but I dropped it since there weren't any references to it yet.

The browserfy thing, I think I got lucky on 😜  I don't know too much about how grunt works. I just found the grunt-browserfy package and it worked.

The problem with the parser was that it wasn't sending anything back to the client. I think it wasn't piping to the parser until I moved the pipe.

thomaswilburn commented:
I'm currently running this live on a Node 4.2.1 instance, so I'm not aware of any changes to the Streams interface that would have happened between Node v4 and v5 affecting the parser pipe. That seems odd, and I would need more information before making that change.

I'm removing the SQLite3 dependency, since we don't use it. It also crashed on build for me (CentOS 6 VM), as did the native module for Postgres (but that has a pure JS fallback and still works, which is excellent). This was also a good reminder for me to add a valid license type, and clean up some of the other dependencies that we don't actually use any more. These changes went in as c27a0903e56108a766d7bcdd3db130400952969e.

This was my first big Node project, and so aspects of the code are probably a bit wonky. If you run into anything else, I'd be happy to hear about it, but be aware that it's been mostly a weird hobby project for me, so the UI and the server itself are idiosyncratic at best.
